app/src/main/java/umc/OnAirMate/ui/pop_up/ChangeNicknameViewModel.kt

```
@@ -0,0 +1,30 @@
+) : ViewModel() {
+
+    // 닉네임 중복 확인 결과를 콜백으로 전달하거나 LiveData로 관리 가능
+    fun checkNickname(
+        nickname: String,
+        onResult: (Boolean) -> Unit
+    ) {
+        viewModelScope.launch {
+            try {
+                val isDuplicated = nicknameRepository.isNicknameDuplicated(nickname)
+                onResult(isDuplicated)
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)